core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -753,21 +749,19 @@ class HttpGoogleServicesDAO(
     } )
   }
 
-  override def getGenomicsOperation(jobId: String): Future[Option[JsObject]] = {
-
-    import spray.json._
-
+  override def getGenomicsOperation(jobId: String): Future[JsObject] = {
     val opId = s"operations/$jobId"
 
     val genomicsApi = new Genomics.Builder(httpTransport, jsonFactory, getGenomicsServiceAccountCredential).setApplicationName(appName).build()
     val operationRequest = genomicsApi.operations().get(opId)
 
-    retryWithRecoverWhen500orGoogleError[Option[JsObject]](() => {
+    retryWithRecoverWhen500orGoogleError[JsObject](() => {
       // Google library returns a Map[String,AnyRef], but we don't care about understanding the response
       // So, use Google's functionality to get the json string, then parse it back into a generic json object
-      Some(executeGoogleRequest(operationRequest).toPrettyString.parseJson.asJsObject)
-    }) {
-      case t: HttpResponseException if t.getStatusCode == StatusCodes.NotFound.intValue => None
+      executeGoogleRequest(operationRequest).toPrettyString.parseJson.asJsObject
+    }) { case e: GoogleJsonResponseException =>
+      val statusCode = StatusCodes.getForKey(e.getStatusCode).getOrElse(StatusCodes.InternalServerError)
```

Comment:
This propagates the status code + message from the Google HTTP call to the end user.